.github/workflows/agent-bundle-linux.yml

```diff
@@ -0,0 +1,33 @@
+name: agent-bundle-linux
+
+on:
+  workflow_call:
+    inputs:
+      ARCH:
+        required: true
+        type: string
+env:
+  GO_VERSION: 1.23.10
+
+jobs:
+  agent-bundle-linux:
+    runs-on: ${{ fromJSON('["ubuntu-24.04", "otel-arm64"]')[inputs.ARCH == 'arm64'] }}
```

Comment:
not related to the main goal of the PR but can we run this on ubuntu arm instead of our runner?

Reply:
Good point. I'll update here, but if anything breaks as a result I'll wait to make the change in a separate PR. 👍